.github/workflows/gradle.yml

```diff
@@ -1,27 +1,51 @@
-name: Build with Gradle
+name: Build Gradle
 
-on: [ push, pull_request ]
 
-permissions:
-  contents: read
```

Comment:
Why remove this?

Reply:
It's useless.

Reply:
It enforces [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) though, so I'd like to keep it